### DISCUSSION TITLE
ignoring invalid defined png:compression-filter =6/7/8/9 @ warning/png.c/WritePNGImage/

### DISCUSSION THREAD

**Discussion by mner8** [ImageMagick/ImageMagick#6108] [Help]
### ImageMagick version

7.1.0-(38-62) Q16 x64

### Operating system, version and so on

Windows 7 SP1 x64

### Description

Why do messages appear when changing PNG compression:

ignoring invalid defined png:compression-filter =6 @ warning/png.c/WritePNGImage/11970.
ignoring invalid defined png:compression-filter =7 @ warning/png.c/WritePNGImage/11970.
ignoring invalid defined png:compression-filter =8 @ warning/png.c/WritePNGImage/11970.
ignoring invalid defined png:compression-filter =9 @ warning/png.c/WritePNGImage/11970.

Bat file example:

```
for /L %%S in (0,1,4) do (
	for /L %%L in (0,1,9) do (
		for /L %%F in (0,1,9) do (
			magick %1 ^
			-define png:compression-strategy=%%S -define png:compression-level=%%L -define png:compression-filter=%%F ^
			%~n1_%%S_%%L_%%F%~x1
		)
	)
)
```

**Comment by urban-warrior** [comment]
Thanks for the problem report.  We can reproduce it and will have a patch to fix it in the GIT main branch @ https://github.com/ImageMagick/ImageMagick later today.  The patch will be available in the beta releases of ImageMagick @ https://imagemagick.org/archive/beta/ by sometime tomorrow.

**Comment by mner8** [comment]
I'm not familiar with PNG encoder. Can someone explain why this message appears?
Why are png:compression-filter=6-9 arguments needed?

**Comment by landfillbaby** [comment]
https://imagemagick.org/script/defines.php says the following:
> png:compression-filter=value
> Valid values are 0 through 9. 0-4 are the corresponding PNG filters, 5 means adaptive filtering except for images with a colormap, 6 means adaptive filtering for all images, 7 means MNG "loco" compression, 8 means Z_RLE strategy with adaptive filtering, and 9 means Z_RLE strategy with no filtering.

So maybe a new check needs to be added so that MNG compression (7) isn't used on regular (A)PNG images?
